core/src/test/java/kafka/server/share/SharePartitionTest.java

```diff
@@ -7134,6 +7137,23 @@ public void testFetchAbortedTransactionRecordBatchesForAbortedAndCommittedTransa
         assertEquals(1, actual.get(3).producerId());
     }
 
+    @Test
+    public void testFetchLockReleasedByDifferentId() {
+        SharePartition sharePartition = SharePartitionBuilder.builder()
+            .withState(SharePartitionState.ACTIVE)
+            .build();
+        Uuid fetchId1 = mock(Uuid.class);
+        Uuid fetchId2 = mock(Uuid.class);
+
+        // Initially, fetch lock is not acquired.
+        assertNull(sharePartition.fetchLock());
+        // fetchId1 acquires the fetch lock.
+        assertTrue(sharePartition.maybeAcquireFetchLock(fetchId1));
+        // If we release fetch lock by fetchId2, it will work.
+        sharePartition.releaseFetchLock(fetchId2);
+        assertNull(sharePartition.fetchLock()); // Fetch lock has been released.
```

Comment:
Please write comments her, why this currently releases the lock. And once we make the locks handling strict then this test case need to be updated.